doc/evaluation_design.md

```diff
@@ -0,0 +1,64 @@
+# Design for ElasticDL Evaluation
+
+The document describes the ElasticDL evaluation process design.
+
+## The Existing Evaluation Design
+
+The master starts to evaluate model every many steps defined by `eval_steps` or every many seconds defined by `eval_secs`. When an evaluation job starts, the master creates evaluation tasks and inserts them into todo queue at head. Then the worker pulls an evaluation task after completing the current task. Because evaluation tasks are located at the head of todo queue.\
+The worker pulls model from parameter serve (pserver) and calculates model prediction outputs with records in the evaluation task, then it reports outputs to master. The evaluation sequence is shown as:
+
+![evaluate_sequence](/doc/figures/evaluate_sequence.svg)
+ <em>Figure 1 </em>. ElasticDL Evaluate Sequence
+
+In order to speed up training, the worker creates `tf.data.Dataset` by [tf.data]((https://www.tensorflow.org/guide/data_performance)) API to construct an input pipeline to load data. The flowchart of train loop and evluation in worker is shown as:
```

Comment:
Be consistent about the word: `train` and `training`. Also a typo here. *evluation*.